docs/TensorLayout.md

```diff
@@ -0,0 +1,219 @@
+## Tensor Layout
+
+This document describes the design of the tensor Layout requirements in Glow.
+
+Certain operations (e.g. convolutions, gemms, etc) need to know the semantic
+layout of their tensors, i.e. the logical ordering ordering of their dimensions
+(e.g. `NHWC`). Some backends enforce additional backend-specific requirements
+on said operations (e.g. tensor alignment).
+
+A theoretical clever backend, might even go a step further and have said
+layout requirements depend on the properties of the operation: a convolution
+with a small filter may need the input operands in a format different from a
+convolution with a big filter.
+
+Tensor layout is a property of the operation, some operations, such as
+element-wise operations, may not care about their input layout, we avoid adding
+a layout field for said operations to reduce the dynamic memory consumption of
+the compiler.
+
+For operations that do have layout requirements, Glow has an easily extendable
+string-based layout field. This allows backends to override Glow's default
+requirements without the hassle of creating a custom, backend-specific, node.
+
+Glow's string-based layout format is encoded as follows:
+
+1. A mandatory one character representing the current dimension. Either an  alphabetic letter or `*` (any layout).
```

Comment:
Please add a concrete example with `*` somewhere. It should be clear which layouts are compatible and what are the rules.

Reply:
I added a new parsing test case for `*` AND I updated the document to explain that :)